src/engines/Cesium/hooks.ts

```diff
@@ -728,6 +733,49 @@ export default ({
     unmountCamera?.();
   }, [unmountCamera]);
 
+  const updateCredits = useCallback(() => {
+    if (!onCreditsUpdate) return;
+    // currently we don't have a proper way to get the credits update event
+    // wait for 3 seconds to get latest credits
+    // some internal property is been used here.
+    setTimeout(() => {
+      const creditDisplay = cesium.current?.cesiumElement?.creditDisplay as
+        | (CreditDisplay & {
+            _currentFrameCredits: {
+              lightboxCredits: { _array: { credit?: CesiumCredit }[] };
+              screenCredits: { _array: { credit?: CesiumCredit }[] };
+            };
+            _currentCesiumCredit: CesiumCredit;
+          })
+        | undefined;
+
+      if (!creditDisplay) return;
+
+      const { lightboxCredits, screenCredits } = creditDisplay?._currentFrameCredits || {};
+      const cesiumCredits = creditDisplay._currentCesiumCredit;
+
+      const credits: Credit[] = [
+        ...(cesiumCredits?.html ? [{ html: cesiumCredits.html }] : []),
+        ...Array.from(lightboxCredits?._array ?? []).map(c => ({
+          html: c?.credit?.html,
+        })),
+        ...Array.from(screenCredits?._array ?? []).map(c => ({
+          html: c?.credit?.html,
+        })),
+      ];
+
+      onCreditsUpdate(credits);
+    }, 3000);
```

Comment:
I can't find a good time point to update the credit since it seems base on load tiles or terrain's meta, for now i just use a hardcoded 3000ms delay.